### PR TITLE
Add get_readiness tool to retrieve status of all registered components

### PR DESCRIPTION
**Pull request by lukekim**
Add a new tool `get_readiness` to retrieve the status of all registered datasets, models, embeddings, views, and other primitives.

* **New Tool Implementation**
  - Add `get_readiness` tool in `crates/runtime/src/tools/builtin/get_readiness.rs` to retrieve the status of all registered components from `RuntimeStatus`.
  - Implement the `get_readiness` tool to use the `RuntimeStatus` struct to get the status of all components and return it.

* **Module Changes**
  - Add `get_readiness` module in `crates/runtime/src/tools/builtin/mod.rs` and include it in the list of available tools.

* **Runtime Status Changes**
  - Add a new method `get_all_statuses` to the `RuntimeStatus` struct in `crates/runtime/src/status.rs` that returns the status of all registered components.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/spiceai/spiceai?shareId=dd2a98f2-3d69-4018-8482-371843be1bc2).

### CODE REVIEW

File: crates/runtime/src/tools/builtin/get_readiness.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        Self { runtime_status }
+    }
+}
+
+#[async_trait]
+impl SpiceModelTool for GetReadinessTool {
+    fn spec(&self) -> ToolSpec {
+        ToolSpec {
+            name: "get_readiness".to_string(),
+            description: "Retrieves the readiness status of all registered datasets, models, embeddings, views, and other primitives.".to_string(),
```
Comment (lukekim) [inline crates/runtime/src/tools/builtin/get_readiness.rs]:
```suggestion
            description: "Retrieves the readiness status of all runtime components including registered datasets, models, and embeddings.".to_string(),
```

**Comment by eadgbear** [comment]
Is the `description` made available to the users in any way? And are `views` always ready? Or do they propagate the underlying source's readyness

File: crates/runtime/src/tools/builtin/get_readiness.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            name: "get_readiness".to_string(),
+            description: "Retrieves the readiness status of all runtime components including registered datasets, models, and embeddings.".to_string(),
+            ..Default::default()
+        }
+    }
+
+    async fn execute(&self, _params: HashMap<String, Value>) -> Result<ToolExecutionResult, ToolExecutionError> {
+        let statuses = self.runtime_status.get_all_statuses();
+        Ok(ToolExecutionResult::new(statuses))
+    }
```
Comment (phillipleblanc) [inline crates/runtime/src/tools/builtin/get_readiness.rs]: Missing quite a few trait implementations:

```rust
pub trait SpiceModelTool: Sync + Send {
    fn name(&self) -> &str;
    fn description(&self) -> Option<&str>;
    fn strict(&self) -> Option<bool> {
        None
    }
    fn parameters(&self) -> Option<Value>;
    async fn call(
        &self,
        arg: &str,
        rt: Arc<Runtime>,
    ) -> Result<Value, Box<dyn std::error::Error + Send + Sync>>;
}
```

Also its `call` not `execute` and returns a `Value`

File: crates/runtime/src/tools/builtin/get_readiness.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    fn parameters(&self) -> Option<Value> {
+        None
+    }
+
+    async fn call(
+        &self,
+        _arg: &str,
+        rt: Arc<Runtime>,
+    ) -> Result<Value, Box<dyn std::error::Error + Send + Sync>> {
+        tracing::span!(target: "task_history", tracing::Level::INFO, "tool_use::get_readiness", tool = self.name());
```
Comment (phillipleblanc) [inline crates/runtime/src/tools/builtin/get_readiness.rs]: Need to add the output as well

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**